File: ClothingStore/wwwroot/js/Product.js
$(document).ready(function () {
    LoadProductGrid();
});

function LoadProductGrid() {
    $("#ProductGrid").load("/Product/ProductGrid");
}


function OpenProductPopup() {

    $("#_CreateProduct").load("/Product/Create", function () {

        $("#productPopup").css("display", "flex");

    });

}
function DeleteProduct(id) {

    $.ajax({

        url: "/Product/DeleteProduct?id=" + id,

        type: "DELETE",

        success: function (response) {

            if (response.success) {

                alert(response.message);

                LoadProductGrid();

            }
            else {

                alert(response.message);

            }

        },

        error: function (xhr) {

            console.log(xhr.responseText);
            alert("Something went wrong.");

        }

    });

}
function CloseProductPopup() {

    $("#productPopup").hide();

}
function AddProduct() {

    var formData = new FormData();

    formData.append("intCategoryId",
        $("#intCategoryId").val()
    );

    formData.append("varName",
        $("#varName").val()
    );

    formData.append("varProductCode",
        $("#varProductCode").val()
    );

    formData.append("varDescription",
        $("#varDescription").val()
    );

    formData.append("flPrice",
        $("#flPrice").val()
    );

    formData.append("flDiscountPrice",
        $("#flDiscountPrice").val()
    );

    formData.append("intQuantity",
        $("#intQuantity").val()
    );

    formData.append("varBrand",
        $("#varBrand").val()
    );

    formData.append("varSize",
        $("#varSize").val()
    );

    formData.append("varColor",
        $("#varColor").val()
    );


    var file = $("#ImageFile")[0].files[0];

    if (file) {
        formData.append("ImageFile", file);
    }


    formData.append("isFeatured",
        $("#isFeatured").is(":checked")
    );

    formData.append("isActive",
        $("#isActive").is(":checked")
    );


    $.ajax({

        url: "/Product/AddProduct",

        type: "POST",

        data: formData,

        contentType: false,

        processData: false,


        success: function (response) {

            alert(response.message);

            LoadProductGrid();

        },


        error: function (xhr) {

            console.log(xhr.responseText);

        }

    });

}


function OpenEditProductPopup(id) {

   
    $("#_EditProduct").load("/Product/Edit/" + id, function () {

        $("#productEditPopup").css("display", "flex");

    });

}



function CloseEditProductPopup() {

    $("#productEditPopup").hide();

}
function UpdateProduct() {


    var formData = new FormData();


    formData.append("intSeqId",
        $("#intSeqId").val());


    formData.append("intCategoryId",
        $("#intCategoryId").val());


    formData.append("varName",
        $("#varName").val());


    formData.append("flPrice",
        $("#flPrice").val());


    formData.append("intQuantity",
        $("#intQuantity").val());



    var file = $("#ImageFile")[0].files[0];


    if (file) {
        formData.append("ImageFile", file);
    }



    $.ajax({

        url: "/Product/UpdateProduct",

        type: "POST",

        data: formData,

        contentType: false,

        processData: false,


        success: function (response) {

            alert(response.message);

            $("#productEditPopup").hide();

            LoadProductGrid();

        }

    });


}

File: ClothingStore/Models/Product.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace ClothingStore.Models
{
    [Table("tblProducts")]
    public class Product
    {
        
            [Key]
            public int intSeqId { get; set; }

            public int? intCategoryId { get; set; }

            [StringLength(250)]
            public string? varName { get; set; }

            [StringLength(50)]
            public string? varProductCode { get; set; }

            public string? varDescription { get; set; }
        [NotMapped]
        public IFormFile? ImageFile { get; set; }
        [Column(TypeName = "decimal(18,2)")]
            public decimal? flPrice { get; set; }

            [Column(TypeName = "decimal(18,2)")]
            public decimal? flDiscountPrice { get; set; }

            public int? intQuantity { get; set; }

            [StringLength(100)]
            public string? varBrand { get; set; }

            [StringLength(50)]
            public string? varSize { get; set; }

            [StringLength(50)]
            public string? varColor { get; set; }

            [StringLength(500)]
            public string? varImageUrl { get; set; }

            public bool? isFeatured { get; set; }

            public bool? isActive { get; set; }

            public DateTime? dtCreatedDate { get; set; }

            public DateTime? dtUpdatedDate { get; set; }

            // Navigation Property
            [ForeignKey("intCategoryId")]
            public virtual Category? Category { get; set; }
     

}
}
